.github/workflows/container.yml

```diff
@@ -30,6 +32,17 @@ jobs:
           registry: ghcr.io
           username: basemachina
           password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Authenticate to Google Cloud
+        uses: 'google-github-actions/auth@v0'
+        id: auth
+        with:
+          workload_identity_provider: ${{ secrets.WORKLOAD_IDENTITY_PROVIDER }}
+          service_account: ${{ secrets.SERVICE_ACCOUNT }}
+      - uses: docker/login-action@v1
+        with:
+          registry: gcr.io
+          username: oauth2accesstoken
+          password: ${{ steps.auth.outputs.access_token }}
```

Comment:
https://github.com/google-github-actions/auth#authenticating-to-container-registry-and-artifact-registry